introspection-engine/connectors/sql-introspection-connector/src/re_introspection.rs

```diff
@@ -0,0 +1,403 @@
+            let fields_to_be_changed = new_data_model.find_enum_fields(&changed_enum_name.0.enm);
+
+            for change2 in fields_to_be_changed {
+                let field = new_data_model.find_field_mut(&change2.0, &change2.1).unwrap();
+                field.field_type = FieldType::Enum(changed_enum_name.1.clone());
+            }
+        }
+    }
+
+    // @map on enum values
+    let mut changed_enum_values = vec![];
+    {
+        for enm in &new_data_model.enums {
+            if let Some(old_enum) = old_data_model.find_enum(&enm.name) {
+                for value in &enm.values {
+                    if let Some(old_value) =
+                        old_enum.find_value_db_name(value.database_name.as_ref().unwrap_or(&value.name.to_owned()))
+                    {
+                        if enm.find_value(&old_value.name).is_none() {
+                            changed_enum_values.push((
+                                EnumAndValue {
+                                    enm: enm.name.clone(),
+                                    value: value.name.clone(),
+                                },
+                                old_value.name.clone(),
+                            ))
+                        }
+                    }
+                }
+            }
+        }
+        for changed_enum_value in &changed_enum_values {
+            let enm = new_data_model.find_enum_mut(&changed_enum_value.0.enm).unwrap();
+            let value = enm.find_value_mut(&changed_enum_value.0.value).unwrap();
+            value.name = changed_enum_value.1.clone();
+            if value.database_name.is_none() {
+                value.database_name = Some(changed_enum_value.0.value.clone());
+            }
+        }
+
+        for changed_enum_value in &changed_enum_values {
+            let fields_to_be_changed = new_data_model.find_enum_fields(&changed_enum_value.0.enm);
+
+            for field in fields_to_be_changed {
+                let field = new_data_model.find_field_mut(&field.0, &field.1).unwrap();
+                if field.default_value
+                    == Some(DefaultValue::Single(PrismaValue::Enum(
+                        changed_enum_value.0.value.clone(),
+                    )))
+                {
+                    field.default_value = Some(DefaultValue::Single(PrismaValue::Enum(changed_enum_value.1.clone())));
+                }
+            }
+        }
+    }
+
+    //virtual relationfield names
+    let mut changed_relation_field_names = vec![];
+    {
+        for model in &new_data_model.models {
+            for field in &model.fields {
+                if let FieldType::Relation(info) = &field.field_type {
+                    if let Some(old_model) = old_data_model.find_model(&model.name) {
+                        for old_field in &old_model.fields {
+                            if let FieldType::Relation(old_info) = &old_field.field_type {
+                                if old_info == info {
```

Comment:
I guess this is where the PartialEq impl without relation name on RelationInfo comes from.